docs/msi_v2/msi_with_credential_design.md

```diff
@@ -0,0 +1,324 @@
+# MSAL MSI V2 /credential Endpoint Design Document
+
+## Overview
+
+This document provides detailed guidance for SDK developers to implement MSI V2 `/credential` endpoint support. It focuses on the **token acquisition process**, ensuring seamless interactions with Managed Identity Resource Providers (MIRPs) on **Azure Virtual Machines (VMs) and Virtual Machine Scale Sets (VMSS)**.
+
+## Goals
+
+The primary objective is to enable seamless token acquisition in MSI V2 for VM/VMSS, utilizing the `/credential` endpoint.
+
+- Define the **MSI V2 token acquisition process**.
+- Describe how MSAL interacts with the `/credential` and the ESTS regional token endpoint.
+- Ensure compatibility with **Windows and Linux** VMs and VMSS.
+
+## Token Acquisition Process
+
+In **MSI V1**, IMDS or any other Managed Identity Resource Provider (MIRP) directly returns an **access token**. However, in **MSI V2**, the process involves two steps:
+
+```mermaid
+sequenceDiagram
+    participant Application
+    participant MSAL
+    participant IMDS
+    participant ESTS
+
+    Application ->> MSAL: 1. Request token using Managed Identity
+    MSAL ->> IMDS: 2. Probe for `/credential` endpoint availability
+    IMDS -->> MSAL: 3. Response (200 OK / 404 Not Found)
+
+    alt `/credential` endpoint available
+        MSAL ->> IMDS: 4. Request Short-Lived Credential (SLC) via `/credential`
+        IMDS -->> MSAL: 5. Return SLC
+        MSAL ->> ESTS: 6. Exchange SLC for Access Token via MTLS
+        ESTS -->> MSAL: 7. Return Access Token
+        MSAL ->> Application: 8. Return Access Token
+    else `/credential` endpoint not available
+        MSAL ->> IMDS: 4a. Fallback to legacy `/token` endpoint
+        IMDS -->> MSAL: 5a. Return Access Token
+        MSAL ->> Application: 6a. Return Access Token
+    end
+```
+
+### Short-Lived Credential Retrieval from `/credential` Endpoint
+
+- Azure Managed Identity Resource Providers host the `/credential` endpoint.
+- The client (MSAL) calls the `/credential` endpoint to retrieve a **short-lived credential (SLC)**.
+- This credential is valid for a short duration and must be used promptly in the next step.
+
+### Access Token Acquisition via ESTS
+
+- The client presents the **short-lived credential** to **ESTS** over **MTLS** as an assertion.
+- ESTS validates the credential and issues an **access token**.
+- The access token is then used to authenticate with Azure services.
+
+## Certificate Handling
+
+To start the flow, MSAL requires a certificate. MSAL follows these steps:
+
+1. **Check for an existing certificate (Windows only)**: MSAL looks for a platform certificate (`devicecert.mtlsauth.local`) in the given Azure resource (In both local machine and local user store). 
```

Comment:
You say "in the given Azure resource". But really, this will only be IMDS, right? I think saying "IMDS" is clearer.

Reply:
As of now the feature is being developed in Vrtual machine, there are other Azure Resources that host IMDS that will not move to use the new /credential route. Hence mentioning Azure Resource would be a better way.